python/langsmith/evaluation/evaluator.py

```diff
@@ -724,3 +706,90 @@ def wrapper(run: Run, example: Example) -> _RUNNABLE_OUTPUT:
                 else wrapper.__name__
             )
             return wrapper  # type: ignore[return-value]
+
+
+def _format_evaluator_result(
+    result: Union[EvaluationResults, dict, str, int, bool, float, list],
+) -> Union[EvaluationResults, dict]:
+    if isinstance(result, (bool, float, int)):
+        result = {"score": result}
+    elif not result:
+        raise ValueError(
+            f"Expected a non-empty dict, str, bool, int, float, list, "
+            f"EvaluationResult, or EvaluationResults. Got {result}"
+        )
+    elif isinstance(result, list):
+        if not all(isinstance(x, dict) for x in result):
+            raise ValueError(
+                f"Expected a list of dicts or EvaluationResults. Received {result}."
+            )
+        result = {"results": result}  # type: ignore[misc]
+    elif isinstance(result, str):
+        result = {"value": result}
+    elif isinstance(result, dict):
+        pass
+    else:
+        raise ValueError(
+            f"Expected a dict, str, bool, int, float, list, EvaluationResult, or "
+            f"EvaluationResults. Got {result}"
+        )
+    return result
+
+
+SUMMARY_EVALUATOR_T = Union[
+    Callable[
+        [Sequence[schemas.Run], Sequence[schemas.Example]],
+        Union[EvaluationResult, EvaluationResults],
+    ],
+    Callable[
+        [List[schemas.Run], List[schemas.Example]],
+        Union[EvaluationResult, EvaluationResults],
+    ],
+]
+
+
+def _normalize_summary_evaluator(func: Callable) -> SUMMARY_EVALUATOR_T:
+    supported_args = ("runs", "examples", "inputs", "outputs", "reference_outputs")
+    sig = inspect.signature(func)
+    positional_args = [
+        pname
+        for pname, p in sig.parameters.items()
+        if p.kind in (p.POSITIONAL_OR_KEYWORD, p.POSITIONAL_ONLY)
+    ]
+    if not positional_args or (
+        not all(pname in supported_args for pname in positional_args)
+        and len(positional_args) != 2
+    ):
+        msg = (
+            f"Invalid evaluator function. Must have at least one positional "
+            f"argument. Supported positional arguments are {supported_args}."
```

Comment:
nit, because I'm dumb, would be nice to pass in the incorrect names to the error message. 

Got confused for a minute too long when I passed `input` instead of `inputs`